### PR TITLE
description for annotations

### PR DESCRIPTION
**Pull request by longxf21**

### CODE REVIEW

File: vscode-antimony/src/server/stibium/stibium/analysis.py [hunk 1]
```diff
@@ -430,6 +434,59 @@ def handle_annotation(self, scope: AbstractScope, annotation: Annotation):
...
+                    definition = res.definition
+                    queried = '\n{}\n\n{}\n'.format(name, definition)
+                    symbol[0].queried_annotations[uri] = queried
+                else:
+                    return
+                    # uniport = UniProt()
+            elif website == 'www.rhea-db.org':
+                rhea = Rhea()
+                df_res = rhea.query(uri_split[4], columns="equation", limit=10)
+                equation = df_res['Equation']
```
Comment (joseph-hellerstein) [inline vscode-antimony/src/server/stibium/stibium/analysis.py:467]: Columns in dataframes should be defined at the top of the module as global constants.

File: vscode-antimony/src/server/stibium/stibium/analysis.py [hunk 2]
```diff
@@ -430,6 +434,59 @@ def handle_annotation(self, scope: AbstractScope, annotation: Annotation):
...
+                queried = '\n{}\n'.format(equation[0])
+                df_res += queried
+                symbol[0].queried_annotations[uri] = queried
+            else:
+                ontology_name = uri_split[-1].split('_')[0].lower()
```
Comment (joseph-hellerstein) [inline vscode-antimony/src/server/stibium/stibium/analysis.py:472]: You should assign the elements of a split to variables. Use of list index of splits is a common source of bugs.

File: vscode-antimony/src/server/stibium/stibium/analysis.py [hunk 3]
```diff
@@ -430,6 +434,59 @@ def handle_annotation(self, scope: AbstractScope, annotation: Annotation):
...
+                iri = uri_split[-1]
+                try:
+                    response = requests.get('http://www.ebi.ac.uk/ols/api/ontologies/' + ontology_name + '/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252F' + iri).json()
```
Comment (joseph-hellerstein) [inline vscode-antimony/src/server/stibium/stibium/analysis.py:475]: Use global constants for URLs.

File: vscode-antimony/src/server/stibium/stibium/symbols.py [hunk 4]
```diff
@@ -211,7 +215,55 @@ def help_str(self):
...
+                    continue
+                if uri in self.queried_annotations.keys():
+                    ret += self.queried_annotations[uri]
+                    continue
+                uri_split = uri.split('/')
+                website = uri_split[2]
+                if website == 'identifiers.org':
+                    if uri_split[3] == 'chebi':
+                        chebi = ChEBI()
+                        res = chebi.getCompleteEntity(uri_split[4])
```
Comment (joseph-hellerstein) [inline vscode-antimony/src/server/stibium/stibium/symbols.py:231]: same comment about splits. Assign the values of the split tuple to variables.

**Review by joseph-hellerstein — COMMENTED** [review]
Themes:
1. There are many cases in which a literal is used (e.g., "equation") in an expression. You should create an upper case variable name (global constant), like EQUATION = "equation" and then use the global constant. This avoids bugs if there is a typo in the literal.
2. Assign variables to the elements of a tuple. split is a tuple. Don't use split[4]. Instead, assign a variable to split[4] and use the variable. This improves readability and simplifies future code modifications if the indices in the tuple change.

**Review by luciansmith — APPROVED** [review]

**Pull request merged**